faidx.c

```diff
@@ -73,26 +119,26 @@ int faidx_main(int argc, char *argv[])
 
     while ( ++optind<argc && exit_status == EXIT_SUCCESS)
     {
-        printf(">%s\n", argv[optind]);
-        int seq_len;
-        char *seq = fai_fetch(fai, argv[optind], &seq_len);
-        if ( seq_len < 0 ) {
-            fprintf(stderr, "Failed to fetch sequence in %s\n", argv[optind]);
-            exit_status = EXIT_FAILURE;
-            break;
-        }
-        size_t i, seq_sz = seq_len;
-        for (i=0; i<seq_sz; i+=60)
-        {
-            size_t len = i + 60 < seq_sz ? 60 : seq_sz - i;
-            if (fwrite(seq + i, 1, len, stdout) < len ||
-                putchar('\n') == EOF) {
-                print_error_errno("faidx", "failed to write output");
-                exit_status = EXIT_FAILURE;
-                break;
-            }
+    /**
+     * This checks if argv[optind] is a file containing regions. The format
+     * needs to be the same, i.e. space delimited. This allows for a large
+     * number of regions. A file or a region can be passed on the command line.
+     * This maintains the UI and solves the "Argument too long" issue.
+     */
+      if( access( argv[optind], F_OK ) != -1 ) {
+        FILE * fn = fopen(argv[optind], "r");
+        char * regions  = read_string_from_fp(fn, 2000);
+        fclose(fn);
+
+        char* token = strtok(regions, " ");
```

Comment:
Hi  Zev,
`strtok` is harmful, you shouldn't use it because you have no guarantee that `print_fasta` will not call `strtok` and change your internal strtok state.

Reply:
Hi Pierre, 
I can see how the static pointer could cause trouble down the line. Would you support the use of strtok_r? Thanks for noticing the potential problem.